### PR TITLE
Disable all MATS telemetry if not configured

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fix for https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1912

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs [hunk 1]
```diff
@@ -93,6 +102,71 @@ public void TelemetryInternalApiSample()
...
+
+            _telemetryManager = new TelemetryManager(
+                _serviceBundle.Config,
+                _platformProxy,
+                null,
+                true);
+
+            var correlationId = Guid.NewGuid().AsMatsCorrelationId();
+
+            var e1 = new ApiEvent(_logger, _crypto, correlationId) { Authority = new Uri("https://login.microsoftonline.com"), AuthorityType = "Aad" };
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs:117]: nit: I think we already have constant values for these

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs [hunk 2]
```diff
@@ -93,6 +102,71 @@ public void TelemetryInternalApiSample()
...
+
+                harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(TestConstants.AuthorityCommonTenant);
+
+                // Act
+                await app
+                    .AcquireTokenInteractive(TestConstants.s_scope)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                // Assert - NoDeviceIdProxy would fail if InternalGetDeviceId is called
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs:152]: there is no assert here?

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs (hunk 1 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs:117]: I'm sure we do, but the tests in this file do not use them. It's just an URI, has no meaning to telemetry.

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs (hunk 2 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/TelemetryTests.cs:153]: Refactored to make assert more clear

**Pull request merged**